extensions/stackdriver/stackdriver.cc

```diff
@@ -151,12 +151,43 @@ std::string getMonitoringEndpoint() {
   return monitoring_endpoint;
 }
 
+// Get GCP project number.
+std::string getProjectNumber() {
+  std::string project_number;
+  if (!getValue({"node", "metadata", "PLATFORM_METADATA", kGCPProjectNumberKey},
+                &project_number)) {
+    return "";
+  }
+  return project_number;
+}
+
 void clearTcpMetrics(::Wasm::Common::RequestInfo& request_info) {
   request_info.tcp_connections_opened = 0;
   request_info.tcp_sent_bytes = 0;
   request_info.tcp_received_bytes = 0;
 }
 
+// Get local node metadata. If mesh id is not filled or does not exist,
+// fall back to default format `proj-<project-number>`.
+void getLocalNodeMetadata(google::protobuf::Struct* node_metadata) {
+  if (!getMessageValue({"node", "metadata"}, node_metadata)) {
+    return;
+  }
+  const auto mesh_id_it = node_metadata->fields().find("MESH_ID");
+  if (mesh_id_it != node_metadata->fields().end() &&
+      !mesh_id_it->second.string_value().empty()) {
```

Comment:
Today the meshid will always fallback to trust domain "<projectname>.svc.id.goog", which cannot be recognized by stackdriver.

we need to add one more line, 

```
// Today GCP Monitoring use "proj-<ProjectNumber>" as mesh ID, reuse input if it is a valid mesh ID.
 if (mesh_id_it != node_metadata->fields().end() &&
      !mesh_id_it->second.string_value().empty() &&
      mesh_id_it->second.string_value().<HAS_PREFIX>("proj-")) {
            return;
      }
```

Reply:
Great catch.
I question the useless default: svc.id.goog ?

Reply:
Done.